### ISSUE TITLE
triton_fabric gateway and resolvers documented as optional, yet omitting them errors

### ISSUE THREAD

**Issue by alasdairrr** [terraform-providers/terraform-provider-triton#28]
The triton_fabric gateway and resolvers are documented as being optional, yet omitting them results in errors:

Error executing CreateFabric request: InvalidArgument: property "gateway": must be an IPv4 address
Error executing CreateFabric request: InvalidArgument: property "resolvers": object value found, but a array is required

**Comment by cheapRoc** [comment]
I can confirm this. It looks like some sort of regression through CloudAPI as the documentation (and an old script of mine) mention that they're optional. This is not Terraform related. I'll try to publish a bug upstream. Thanks for the report!

```
* triton_fabric.testnet: Error executing CreateFabric request: InvalidArgument: property "resolvers": object value found, but a array is required
```
```
* triton_fabric.testnet: Error executing CreateFabric request: InvalidArgument: property "gateway": must be an IPv4 address
```

**Comment by cheapRoc** [comment]
Nevermind... this is definitely a regression within the Terraform provider. We're somehow now sending values when we shouldn't be. I can get this patched up tomorrow AM.

**Comment by cheapRoc** [comment]
Looks like adding a few missing `omitempty` on our JSON encoding solves the problem. I'll check other similar cases and submit a patch.

**Comment by stack72** [comment]
@jen20 / @sean- closed via #29
